From fdc2358681d8b1b22531aa3d9e2d2d804646afe1 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Thu, 21 Apr 2022 17:50:43 +0000
Subject: [PATCH] libstdc++: make CredentialLeakDialogController destructor
 public

std::unique_ptr in credential_leak_dialog_factory::CreateController()
calls destructor of CredentialLeakDialogController, but it is
protected.

Bug: 957519
Change-Id: If557058a198a231754af597a39ba4fd78d4eb2a9
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3598892
Reviewed-by: Vasilii Sukhanov <vasilii@chromium.org>
Commit-Queue: Stephan Hartmann <stha09@googlemail.com>
Cr-Commit-Position: refs/heads/main@{#994797}
---
 .../browser/ui/passwords/credential_leak_dialog_controller.h | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/chrome/browser/ui/passwords/credential_leak_dialog_controller.h b/chrome/browser/ui/passwords/credential_leak_dialog_controller.h
index 8c4c34a9c662e..aa740131744be 100644
--- a/chrome/browser/ui/passwords/credential_leak_dialog_controller.h
+++ b/chrome/browser/ui/passwords/credential_leak_dialog_controller.h
@@ -14,6 +14,8 @@
 // the state.
 class CredentialLeakDialogController : public PasswordBaseDialogController {
  public:
+  ~CredentialLeakDialogController() override = default;
+
   // Called when the user cancels the dialog by clicking a button.
   virtual void OnCancelDialog() = 0;
 
@@ -45,9 +47,6 @@ class CredentialLeakDialogController : public PasswordBaseDialogController {
 
   // Checks whether the dialog should show cancel button.
   virtual bool ShouldShowCancelButton() const = 0;
-
- protected:
-  ~CredentialLeakDialogController() override = default;
 };
 
 #endif  //  CHROME_BROWSER_UI_PASSWORDS_CREDENTIAL_LEAK_DIALOG_CONTROLLER_H_
